### PR TITLE
setup ossf scorecard and codql workflows

### PR DESCRIPTION
**Pull request by mmorel-35**
Signed-off-by: Matthieu MOREL <matthieu.morel35@gmail.com>

### CODE REVIEW

**Comment by mmorel-35** [comment]
cc @phlax

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    branches: [ "main" ]
+  schedule:
+    - cron: '41 21 * * 0'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    timeout-minutes: 360
+    permissions:
```
Comment (phlax) [inline .github/workflows/codeql.yml:19]: lets move these up - this were the actual grants happen

**Comment by phlax** [comment]
cc @mum4k

File: .github/dependabot.yml [hunk 2]
```diff
@@ -3,7 +3,15 @@
...
 version: 2
 updates:
-  - package-ecosystem: "pip"
-    directory: "/salvo"
+  - package-ecosystem: github-actions
+    directory: /
     schedule:
-      interval: "weekly"
+      interval: weekly
+  - package-ecosystem: gomod
```
Comment (phlax) [inline .github/dependabot.yml:10]: least indent please

File: .github/workflows/scorecard.yml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      # Needed to upload the results to code-scanning dashboard.
+      security-events: write
+      # Needed to publish results and get a badge (see publish_results below).
+      id-token: write
+      # Uncomment the permissions below if installing in a private repository.
+      # contents: read
+      # actions: read
+
+    steps:
+      - name: "Checkout code"
```
Comment (phlax) [inline .github/workflows/scorecard.yml:30]: same on these

**Review by mum4k — APPROVED** [review]

**Pull request merged**